esmvaltool/cmorizers/data/formatters/datasets/esacci_landcover.py

```diff
@@ -0,0 +1,378 @@
+
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube with fixed coordinates.
+    """
+    # First fix any completely missing coord var names
+    fix_dim_coordnames(cube)
+
+    # Convert longitude from -180...180 to 0...360
+    cube = cube.intersection(longitude=(0.0, 360.0))
+
+    # Fix individual coords
+    for cube_coord in cube.coords():
+        # Fix time
+        if cube_coord.var_name == 'time':
+            logger.info("Fixing time...")
+            cube.coord('time').convert_units(
+                Unit('days since 1950-1-1 00:00:00', calendar='gregorian'))
+
+        # Fix latitude
+        if cube_coord.var_name == 'lat':
+            logger.info("Fixing latitude...")
+            cube = iris.util.reverse(cube, cube_coord)
+
+        # Fix bounds of all coordinates
+        fix_bounds(cube, cube_coord)
+
+    return cube
+
+
+def save_variable(cube, var, out_dir, attrs, **kwargs):
+    """Saver function.
+
+    Saves iris cubes (data variables) in CMOR-standard named files.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Data cube to be saved.
+
+    var: str
+        Variable short_name e.g. ts or tas.
+
+    out_dir: str
+        Root directory where the file will be saved.
+
+    attrs: dict
+        Dictionary holding cube metadata attributes like
+        project_id, version etc.
+
+    **kwargs: kwargs
+        Keyword arguments to be passed to `iris.save`
+    """
+    fix_dtype(cube)
+
+    # Ensure the variable metadata is correctly set
+    cube.var_name = var
+
+    # Only set standard_name if it is a valid CF standard name
+    if hasattr(cube, 'standard_name') and cube.standard_name in \
+            iris.std_names.STD_NAMES:
+        cube.standard_name = cube.standard_name
+    else:
+        cube.standard_name = None  # or keep it unset
+
+    # Ensure long_name is set
+    cube.long_name = cube.long_name or var
+
+    # Debugging metadata before saving
+    logger.debug("Saving cube with var_name=%s, standard_name=%s, "
+                 "long_name=%s",
+                 cube.var_name, cube.standard_name, cube.long_name)
+
+    # CMOR standard
+    try:
+        time = cube.coord('time')
+    except iris.exceptions.CoordinateNotFoundError:
+        time_suffix = None
+    else:
+        year = f"{time.cell(0).point.year:d}"
+        time_suffix = '-'.join([year])
+
+    name_elements = [
+        attrs['project_id'],
+        attrs['dataset_id'],
+        attrs['modeling_realm'],
+        attrs['version'],
+        attrs['mip'],
+        var,
+    ]
+    if time_suffix:
+        name_elements.append(time_suffix)
+    file_name = '_'.join(name_elements) + '.nc'
+    file_path = os.path.join(out_dir, file_name)
+    logger.info('Saving: %s', file_path)
+    status = 'lazy' if cube.has_lazy_data() else 'realized'
+    logger.info('Cube has %s data [lazy is preferred]', status)
+    iris.save(cube, file_path, fill_value=1e20, **kwargs)
```

Comment:
Why do you need a custom `save_variable` function?

Reply:
I needed to manage the standard_name and long_name issue I was encountering during the saving by using the save_variable function in utilities but I might just integrate the related lines 187-198 into the save_variable function in the utilities indeed as the rest of the code is the same

Reply:
I am referring to the `standard_name `of the variables we are interested in that are missing in the raw data downloaded from CEDA and so I was getting an error while saving them after the processing
![image](https://github.com/user-attachments/assets/4675aa1f-9adb-46c6-9069-7b74be029fbb)

Reply:
Right, but after calling [`fix_var_metadata`](https://github.com/ESMValGroup/ESMValTool/blob/f8c673ff83150647d42a147137ab296597caa9cc/esmvaltool/cmorizers/data/utilities.py#L259) (which you do), each file should have a valid standard name.

Reply:
Thanks! I am doing some testing by using  the `save_variable `in utilities as this custom one looks like a duplication indeed.